crates/prune/prune/src/segments/history.rs

```diff
@@ -10,103 +10,158 @@ use reth_db_api::{
 };
 use reth_provider::DatabaseProviderRW;
 
-/// Prune history indices up to the provided block, inclusive.
+enum PruneShardOutcome {
+    Deleted,
+    Updated,
+    Unchanged,
+}
```

Comment:
this cries to be transferred to a metric at end of `prune_history_indices` when returned to scope of segment, could you impl it as part of this pr or make an issue

Reply:
v

Reply:
https://github.com/paradigmxyz/reth/issues/9330